### ISSUE TITLE
Possible contributions for newcomers

### ISSUE THREAD

**Issue by Datseris** [JuliaMusic/JuliaMusic_documentation.jl#10]
Hi there! If you want to contribute in JuliaMusic, there are several directions you can follow:

- [ ] The package https://github.com/JuliaMusic/MusicProcessing.jl has recently been transferred to the org. It targets real audio files, instead of MIDI-like data. However, it needs to be slowly updated to Julia 1.5+. 
- [ ] Once MusicProcessing.jl is updated, we can start thinking about adding new features there. One example is peak-detection or peak-detection from the waveforms. (we can start opening issues there as feature requets after the update)
- [ ] Another package that needs some updating, or at least test-driving is Mplay.jl, which makes a MIDI player based on some C libraries. This has to be updated to Julia 1.5+ as well, and use proper Artifacts.jl as well as the binary builder.
- [ ] The packages MIDI.jl and MusicManipulations.jl are very stable and communicate well. These two packages have their own issue tracker with feature requests, some of which are particularly suited for new comers. For example I al linking:
    * https://github.com/JuliaMusic/MusicManipulations.jl/issues/30 , https://github.com/JuliaMusic/MusicManipulations.jl/issues/56 , https://github.com/JuliaMusic/MIDI.jl/issues/110 
- [ ] MusicXML.jl and MIDI.jl communication can be improved: i.e. we should be able to in principle translate a `MIDIFile` or `Notes` into music xml information (and hopefully even vice versa!)
- [ ] Adding MusicXML.jl and Mplay to the Julia Music docs
- [ ] Ensuring the JuliaMusic docs are appropriately organized for a beginner

And of course more important than anything: new feature requests everywhere!

**Comment by jheinen** [comment]
> Another package that needs some updating, or at least test-driving is Mplay.jl, which makes a MIDI player based on some C libraries. This has to be updated to Julia 1.5+ as well, and use proper Artifacts.jl as well as the binary builder.

I don't think it makes sense to migrate the few functions related to MIDI or console I/O to Julia or to use artifacts for that. I don't think you need to write (or even wrap) such low-level functions in Julia. It would be technically possible, yes, but what would it be good for?

Test-driving is also a problem. Keep in mind that we are talking about a real-time application here. Anyway, I could not perceive possible drop-outs caused by delays in MIDI message transmission due to user interface updates (applies to both GUI and TUI) on current hardware.

I am/was only a hobby musician, but I imagine I have an advantage over a computer here. @Datseris : As a drummer, you should know what I mean. 😀
